### ISSUE TITLE
Unable to clear unread message notifications

### ISSUE THREAD

**Issue by jlfrance** [vector-im/element-web#23754]
### Steps to reproduce

1. I have read all my unread messages except for 15 messages in a room (internal room ID: !uLXVepEhQBObRjarPD:thesis.co)
2. I open the room
3. I click on threads
4. I see two threads with a red dot
5. I click each thread, scroll through the messages, and the unread message notification count changes to 19

#### What did you expect?
I expected when I clicked on the threads with the red dot, that the unread message notifications would change to zero

#### What happened instead?
When I clicked on the threads with the red dot, the unread message count increased

### Operating system

macOS

### Application version

Element version: 1.11.5 Olm version: 3.2.12

### How did you install the app?

https://element.io/get-started#download

### Homeserver

chat.thesis.co

### Will you send logs?

Yes

**Comment by jlfrance** [comment]
A similar issue just happened in room id# !DuyKtxtmoGxbTLZlRW:thesis.co
The room had zero unread message notifications. I clicked into the room, clicked on the threads, and the unread message count for the room changed to 17.

**Comment by gsouquet** [comment]
Thanks for you reporting this issue.
It looks very similar to https://github.com/vector-im/element-web/issues/23683 which has been fixed as part of https://github.com/matrix-org/matrix-js-sdk/pull/2862. 

You can test those changes early on develop.element.io or wait for the next release of Element Web.

**Comment by jlfrance** [comment]
I tested the changes in develope.element.io and the issues with room id# !DuyKtxtmoGxbTLZlRW:thesis.co are resolved however the unread message notifications from room ID: !uLXVepEhQBObRjarPD:thesis.co are not resolved. Additionally, there are a large number of persistent unread notifications appearing in other rooms. Do you suggest I report those or what should my expectation be regarding those?

**Comment by TheCycoONE** [comment]
@jlfrance I'm not sure if this is your issue, but I started using threads and was seeing unread notifications in a room that wouldn't go away. There were also unread notifications in the thread view, but I couldn't see them. I eventually found there were expandable sections within the thread view that once all expanded made the notification count go away.

It was very hard to discover.

**Comment by jlfrance** [comment]
Thanks @TheCycoONE, that was not the issue, I was aware of the "all threads" & "my threads" option. I am always able to locate the unread message, but the notification never clears despite reading each message.

**Comment by jlfrance** [comment]
@gsouquet Thanks, I tested the changes in develope.element.io and the issues with room id# !DuyKtxtmoGxbTLZlRW:thesis.co are resolved however the unread message notifications from room ID: !uLXVepEhQBObRjarPD:thesis.co are not resolved. Additionally, there are a large number of persistent unread notifications appearing in other rooms. Do you suggest I report those or what should my expectation be regarding those?

**Comment by gsouquet** [comment]
@jlfrance reopening this issue as it seems there are problems that are unresolved with it. 

Could you clarify the difference between this issue and https://github.com/vector-im/element-web/issues/23770 ? They appear as duplicate to me, but there might be a subtle difference that I missed?

I would be interested in understanding a bit more the issue with persistent unread notifications? Under what circumstances do those appear (when starting Element up? when opening a thread?) ? 

Are there ways to clear them up, does `All Settings > Notifications > 'Clear notifications'` have any impact?

**Comment by jlfrance** [comment]
@gsouquet Yes, you are correct, this is an identical issue to #23770. I was unable to locate 23770 and thought I had not actually submitted the issue so I submitted another issue. Sorry for that oversight. 

The issue was the unread messages were persistent through the closing and reopening of Element - during the testing 3 days ago, I was able to successfully clear the messages in !DuyKtxtmoGxbTLZlRW:thesis.co however, the messages in !uLXVepEhQBObRjarPD:thesis.co persisted despite opening the indicated threads, closing the app and reopening, etc. Today, when I tested again, I no longer see unread messages in !uLXVepEhQBObRjarPD:thesis.co. However, during testing, I clicked on a room that did not have an unread notification in it which resulted in an unread notification popping up. Attempts to clear this were unsuccessful (screen recording attached) until I follow `All Settings > Help & About > Clear cache & reload`. 

Additionally, `All Settings > Notifications > 'Clear notifications'` has not historically worked (screen recording attached).

To recap: after testing just now, the original notifications WERE cleared, and all other notifications were cleared through `All Settings > Help & About > Clear cache & reload`

**Comment by jlfrance** [comment]
video showing the notification in !DuyKtxtmoGxbTLZlRW:thesis.co won't clear

**Comment by johanricher** [comment]
I can confirm I have this bug too.

Before closing the client, I have no unread messages. When I open it back, some rooms have unread messages (while there has been no actual new message). At the moment, I manage to remove the unread messages by clicking on threads at random to open/load their messages, until the unread counter disappears. This seem to happen with random rooms (I cannot figure the pattern or cause of this bug) and be solved by opening random threads within those rooms.

**Comment by johanricher** [comment]
Can you confirm that the fix to this issue is being used in the current version of element-web (`1.11.16`)? Because I'm still experiencing the issues I shared above.

**Comment by knoedel001** [comment]
I have this issue as well. Not using threads, but several normal direct messages and rooms have stuck notifications. I can clear them temporarily with `All Settings > Help & About > Clear cache & reload`, but they keep reappearing after the next restart of Element. Issue exists on Element Web and Element Desktop on different computers. The mobile clients for Android and iOS have no issues and also Fluffy Chat does not display any notifications left. The problem exists for longer time now and the notifications stuck are getting more and more. My Element Web and Desktop installation is on 1.11.26.

**Comment by aislinnORG** [comment]
We also still have this issue on our Desktop app on Mac laptops! Feels like whack-a-mole. The new "mark as read" function means you can clear the badges but then they just pop back up at random. I am currently on Version 1.11.34 (1.11.34).

**Comment by t3chguy** [comment]
https://github.com/vector-im/element-web/issues/24392 is the issue to track
